back/engines/commercial/insights/app/finders/insights/inputs_finder.rb

```diff
@@ -39,6 +40,20 @@ def filter_category(inputs)
             .where(insights_category_assignments: { category_id: category_id })
     end
 
+    def filter_processed(inputs)
+      return inputs if params[:processed].blank?
+      return inputs unless %w[true false].include?(params[:processed])
+
+      return inputs.left_outer_joins(:insights_processed_flags)
+            .where(insights_processed_flags: { view: view }) if params[:processed] == 'true'
+
+      inputs_with_flags = inputs.left_outer_joins(:insights_processed_flags)
+
+      inputs_with_flags
+        .where.not(insights_processed_flags: { view: view })
+        .or(inputs_with_flags.where(insights_processed_flags: { id: nil }))
```

Comment:
Could something like that work?
```ruby
inputs_with_flags = inputs.left_outer_joins(:insights_processed_flags)
                          .where(insights_processed_flags: { view: [view, nil] })

if params[:processed] == 'true'
	inputs_with_flags.where.not(insights_processed_flags: { id: nil })
else
    inputs_with_flags.where(insights_processed_flags: { id: nil })
end
```

Didn't test it, but I suspect that the `.where.not(insights_processed_flags: { view: view })` will return inputs that were processed but in the context of another view.